### DISCUSSION TITLE
Product discount for some specific collection

### DISCUSSION THREAD

**Discussion by phamquyetthang** [Shopify/function-examples#545] [Feedback: Discounts APIs]
I see that currently the Product Discount API's Input does not allow me to get the collections id and tag id of the product, so I am having difficulty creating a discount app for my purposes.

My case:

Products in collection A will receive a 5% discount.

Products in collection B will receive a 10% discount.

Products that do not belong to these collections but belong to tag C will receive an 8% discount if you buy 2 or more products.

I see that if I use variables in input queries for inAnyCollection, it will filter out all products in a collection from the beginning, so I don't know how I can implement the above logic.

Hope you can help, thank you !

**Comment by ismaArce** [comment]
You can solve this problem by using inCollections instead of inAnyCollection in your input.graphql. Additionally, you need to define your DiscountApplicationStrategy as ALL and specify multiple targets that will receive different discounts. This approach allows us to create multiple discounts that apply all at once using just one form and one DiscountNode. If you need some inspiration, you can check out our app at the following link: https://apps.shopify.com/discount-genie
